src/bootstrap/config.rs

```diff
@@ -889,26 +891,51 @@ impl Default for RustOptimize {
     }
 }
 
-fn deserialize_and_validate_opt_level<'de, D>(d: D) -> Result<String, D::Error>
+fn deserialize_and_validate_opt_level_string<'de, D>(d: D) -> Result<String, D::Error>
 where
     D: serde::de::Deserializer<'de>,
 {
     let v = String::deserialize(d)?;
-    if ["0", "1", "2", "3", "s", "z"].iter().find(|x| **x == v).is_some() {
+    if ["s", "z"].iter().find(|x| **x == v).is_some() {
         Ok(v)
     } else {
-        Err(format!(r#"unrecognized option for rust optimize: "{}", expected one of "0", "1", "2", "3", "s", "z""#, v)).map_err(serde::de::Error::custom)
+        Err(format!(
+            r#"unrecognized option for rust optimize: "{}", expected one of 0, 1, 2, 3, "s", "z""#,
```

Comment:
We should probably include `false` and `true` in the list of allowed values. Both here and when parsing the int. And possibly put the error message into a `const` variable, so that it's not duplicated.

Reply:
Added.